src/pages/workspaces/workspace/Data.js

```diff
@@ -417,32 +422,20 @@ const WorkspaceData = _.flow(
   const signal = useCancellation()
   const asyncImportJobs = useStore(asyncImportJobStore)
 
-
-  // Helpers
-  const getSelectionType = () => {
-    const referenceData = getReferenceData(attributes)
-    return Utils.cond(
-      [!selectedDataType, () => 'none'],
-      [selectedDataType === localVariables, () => 'localVariables'],
-      [selectedDataType === bucketObjects, () => 'bucketObjects'],
-      [_.isArray(selectedDataType), () => 'snapshots'],
-      [_.includes(selectedDataType, _.keys(referenceData)), () => 'referenceData'],
-      () => 'entities'
-    )
-  }
-
   const loadEntityMetadata = async () => {
     try {
       setEntityMetadata(undefined)
       setEntityMetadataError(false)
       const entityMetadata = await Ajax(signal).Workspaces.workspace(namespace, name).entityMetadata()
 
-      setSelectedDataType(getSelectionType() === 'entities' && !entityMetadata[selectedDataType] ? undefined : selectedDataType)
+      if (selectedData?.type === workspaceDataTypes.entities && !entityMetadata[selectedData.entityType]) {
+        setSelectedData(null)
```

Comment:
I have no preference here, but curious why you chose `null` over `undefined`. Maybe pop-in a PR comment right here?

Reply:
Mostly habit I suppose. But looking back, `undefined` would be more consistent with other state variables in this component.

Reply:
Added a utils function in [ec3187e](https://github.com/DataBiosphere/terra-ui/pull/3316/commits/ec3187ec4c7f6b2b2ae8073dccfe8757f2618919).